### DISCUSSION TITLE
signal-store entity selectors

### DISCUSSION THREAD

**Discussion by zmara** [ngrx/platform#4463] [Ideas]
### Which @ngrx/* package(s) are relevant/related to the feature request?

signals

### Information

Alternative to @ngrx/store selectors with parameters
`export const selectTicketById = (id: string) =>
  createSelector(
    selectEntities,
    entities => entities[id]
  );`

Something like:
`
withEntities({entity}),
withComputed(({entities) => ({
   selectById: (id: string) => computed(() => entities[id])
})
`

### Describe any alternatives/workarounds you're currently using

I get all entities from store in a component with computed filter one by id outside of the store withComputed

### I would be willing to submit a PR to fix this issue

- [ ] Yes
- [X] No

**Comment by muuvmuuv** [comment]
Did you came up with a workaround? I also think https://github.com/ngrx/platform/discussions/4634 is a duplicate.

**Comment by christianspiller** [comment]
I solved it with linkedSignal.

    import { linkedSignal, Signal } from "@angular/core";
    ...
    withMethods((store) => ({
      getEntityById(id: string): Signal<MyEntity | undefined> {
        return linkedSignal(
          () => store.entityMap()[id],
          { equal: (a, b) => a === b }
        );
    }

Warning: Status 'developer preview' in Angular 19 (https://v19.angular.dev/guide/signals/linked-signal)
Warning 2: Linked Signals are writeable. Wrap it in an additional computed() if necessary.

**Reply by rainerhahnekamp** [reply]
Since v20, you can also use `withLinkedState` and `linkedSignal` (which it uses under the hood) is also stable now.

**Reply by ydemartino** [reply]
Can you show an example on how to use it with `withLinkedState`?

I tried to use it, but it seems I am not able to pass the id parameter to the function.
Thanks

**Reply by muuvmuuv** [reply]
Did you find out?

**Reply by rainerhahnekamp** [reply]
@muuvmuuv, @ydemartino if you could post a link to a Stackblitz it would be easier to help.
